text/XXXX-config_revision.md

```diff
@@ -0,0 +1,268 @@
+  - /ip4/35.202.155.225/tcp/4001/ipfs/QmegNYmwHUQFc3v3eemsYUVf3WiSg4RcMrh3hovA5LncJ2
+  - /ip4/35.238.10.180/tcp/4001/ipfs/QmessbA6uGLJ7HTwbUJ2niE49WbdPfzi27tdYXdAaGRB4G
+  - /ip4/35.238.105.35/tcp/4001/ipfs/Qmc353gHY5Wx5iHKHPYj3QDqHP4hVA1MpoSsT6hwSyVx3r
+  - /ip4/35.239.138.186/tcp/4001/ipfs/QmT9YHJF2YkysLqWhhiVTL5526VFtavic3bVueF9rCsjVi
+  - /ip4/35.226.44.58/tcp/4001/ipfs/QmQS2ryqZrjJtPKDy9VTkdPwdUSpTi1TdpGUaqAVwfxcNh
+Profile:
+  color: ""
+  created: "2020-02-12T04:29:18Z"
+  description: ""
+  email: brendan+test@qri.io
+  homeurl: ""
+  id: QmSyDX5LYTiwQi861F5NAwdHrrnd1iRGsoEvCyzQMUyZ4W
+  name: ""
+  peername: b5
+  photo: https://qri-user-images.storage.googleapis.com/1570029824064.png
+  poster: /ipfs/QmSN2yHp4qFy1xLCRVQNJKAjgbwHYPAho9fT7C3vAM426c
+  privkey: <base64-encoded private key>
+  thumb: ""
+  twitter: ""
+  type: peer
+  updated: "2018-04-19T18:10:46.627471799-04:00"
+RPC:
+  enabled: true
+  address: "/ip4/0.0.0.0/tcp/2504"
+Registry:
+  location: https://registry.qri.cloud
+Remote:
+  acceptsizemax: 0
+  accepttimeoutms: 0
+  allowremoves: false
+  enabled: false
+  requireallblocks: false
+Remotes: null
+Repo:
+  type: fs
+Revision: 2
+Filesystems:
+- type: ipfs
+  options:
+    path: ./ipfs
+    api: true
+    pubsub: false
+- type: local
+- type: http
+```
+
+### Expanding Store to Filesystems
+
+Currently we hard-code qri filesystem (`qfs`) configuration details. qfs spins up a multiplexed-filesystem by default, composing multiple file-like persistence layers using prefix demuxing. If we expand "Store" to "Filesystem", we can make this fully configurable. The default configuration qri is using would look like this:
+
+```yaml
+Filesystems:
+- type: ipfs
+  config:
+    path: ~/.ipfs
+    api: true
+    pubsub: false
+- type: local
+- type: http
+```
+
+The first win: users could remove `local` or `http` to disable qri's access to the local filesystem or HTTP for the purpose of reading data. We can also expand for later filesystems & options. an `s3` implementation would be nice. this is also how we'd do `dat` integration.
+
+We could use a convention of the default write destination being the first listed filesystem, and present a big warning if users configure anything other than `ipfs` in the first position.
+
+It'd be great to delegate this entire section of store setup to the qfs package, aliasing up the conguration data structure into the config package.
+
+### move IPFS repo location into configuration
+the above filesystem example shows a `ipfs` filesystem type with a `config.path` field. This value should now the be canonical source of where to load.
+
+the global `ipfs-path` flag should also be removed:
+```
+      --ipfs-path string   override IPFS path location (default "/Users/b5/.ipfs")
+```
+
+### Relative Paths in configuration files are relative to the file
+If a configuration value specifies a filepath, and the given path is relative, that path should be relative to the location of the config file.
+
```

Comment:
Agree on this as it resolves ambiguity, however this should be properly documented behavior somewhere.